### ISSUE TITLE
e2e-test-utils-playwright - RequestUtils - add support for extraHTTPHeaders

### ISSUE THREAD

**Issue by Chrico** [WordPress/gutenberg#62466]
## What problem does this address?

Currently the [RequestUtils::setup()](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-test-utils-playwright/src/request-utils/index.ts#L88) creates a new Context and only uses `baseURL` and `storageState` (by resolving the `storageStatePath`), but it does not support [further options](https://playwright.dev/docs/api/class-browser#browser-new-context) like `extraHTTPHeaders` or `httpCredentials`.

## What is your proposed solution?
It would be good to enhance the `RequestUtils::setup()`-method by splitting up the paramaters into "browser context options"-object to support all possible options and the rest (like `user`) is separated. 

So instead we could change it to following:

**Edit:** Changed from `BrowserContextOptions` to `APIRequestContextOptions` as mentioned here: https://github.com/WordPress/gutenberg/issues/62466#issuecomment-2204543331

```ts
interface APIRequestContextOptions extends Readonly< {
	baseURL?: string,
	extraHTTPHeaders?:  { [ key: string ]: string },
	httpCredentials?: {
		username: string,
		password: string,
		origin?: string
	},
	ignoreHTTPSErrors?: boolean,

	// snip
}> {}


class RequestUtils {

	// snip

	static async setup( context?: APIRequestContextOptions, user?: User, storageStatePath?: string ) {
	   // snip
	}
}
```

See full list of APIRequest Context Options: https://playwright.dev/docs/api/class-apirequest#api-request-new-context

**Comment by kopepasah** [comment]
Currently, we have a `test` site that uses `httpCredentials` for access, and we are unable to pass any additional options to the `setup()` method. I agree with @Chrico that these options must be added, and the best case is to pass all available options to Playwright, rather than limit the options available with the WordPress E2E library. 

Generally speaking, when a abstraction library is created, it is a good practice to allow passing of any additional options to the underlying library (unless there is a specific reason not to do so).

Does anyone know if there was a specific reason for not allowing all options to be passed (asking just in case there is a caveat to this approach)?

**Comment by kopepasah** [comment]
> See full list of Browser Context Options: https://playwright.dev/docs/api/class-browser#browser-new-context

@Chrico I think we need to pass the https://playwright.dev/docs/api/class-apirequest#api-request-new-context, correct? Very similar to browser context options, but specific to the [APIRequest](https://playwright.dev/docs/api/class-apirequest) class.

**Comment by Chrico** [comment]
> @Chrico I think we need to pass the https://playwright.dev/docs/api/class-apirequest#api-request-new-context, correct? Very similar to browser context options, but specific to the [APIRequest](https://playwright.dev/docs/api/class-apirequest) class.

Yes, you're right, the `RequestUtils`  is not using the Browser and instead the APIRequest. 👍🏻

**Comment by Chrico** [comment]
@jordesign any feedback regarding that issue? :-)

**Comment by jordesign** [comment]
@Chrico  - thank you, but no feedback - this kind of thing is above my technical expertise :)
